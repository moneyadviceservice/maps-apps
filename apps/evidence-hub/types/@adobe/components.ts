export interface LinkType {
  linkTo: string;
  text: string;
  externalLink?: boolean;
  description: string | null;
}

export type LinkGroup = {
  title: string | null;
  childLinks: LinkType[];
};

export interface Logo {
  image: ImageType;
  altText: string;
}

export interface ImageType {
  _path: string;
  width: number;
  height: number;
  mimeType: string;
}
