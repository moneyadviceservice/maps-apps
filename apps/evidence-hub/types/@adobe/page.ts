import { JsonRichText } from '@maps-react/vendor/utils/RenderRichText';

import { LinkType } from './components';

export interface Tag {
  name: string;
  key: string;
}

export interface DocumentTemplate {
  seoTitle: string;
  seoDescription: string;
  title: string;
  slug: string;
  publishDate: string;
  lastUpdatedDate?: string;
  pageType?: Tag;
  dataType?: Tag[];
  tags?: Tag[];
  breadcrumbs?: LinkType[];
  sections: JsonRichText[];
  contactInformation: JsonRichText;
  clientGroup: Tag[];
  topic: Tag[];
  countryOfDelivery: Tag[];
  links: LinkType[];
  overview?: JsonRichText;
}

export interface TagListItem {
  tagGroup: {
    label: string;
    tags: Tag[];
  };
}
