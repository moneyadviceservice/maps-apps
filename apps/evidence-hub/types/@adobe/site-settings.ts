import { LinkGroup, LinkType, Logo } from './components';

export interface SiteSettings {
  seoTitle: string;
  seoDescription: string;
  headerLogo: Logo;
  mainNavigation: LinkType[];
  footerLinks: LinkGroup[];
}
