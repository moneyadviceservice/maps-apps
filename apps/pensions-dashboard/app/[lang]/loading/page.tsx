import { Metadata, NextPage } from 'next';
import { PensionsDashboardLayout } from '../../../layouts/PensionsDashboardLayout';
import { SITE_TITLE } from '../../../utils/constants';
import { Paragraph } from '@maps-digital/ui/components/Paragraph';
import { Link } from '@maps-digital/ui/components/Link';

const title = 'Pensions found';

export const generateMetadata = async (): Promise<Metadata> => {
  return {
    title: `${title} | ${SITE_TITLE}`,
  };
};

const Page: NextPage = async () => {
  // (1) CDA FIND - 25781
  // > check if data load has at least started, if not complete
  // - use GET not found
  // - or store a cookie when POST has started

  // if NO

  // - create userSessionId UUID
  // - SET cookie
  //   - `userSessionId`
  // - POST to `/api/redirect-details` with
  //   - redirectPurpose = "FIND",
  //   - userSessionId,
  //   - iss = "MHPD-75b68255-444e-4d5f-bbfe-249c26d69963"} (react app env var)
  // - SET cookie, from response
  //   - `codeVerifier`
  // - redirect to CDA emulator
  //   -redirectTargetUrl
  //   - query params from (spread data)
  //   - `redirect_uri` with return URL to React app
  //   - `{redirectTargetUrl}?redirect_uri=http://localhost:4100`

  return (
    <PensionsDashboardLayout title={title}>
      <>
        <Paragraph>Loading...</Paragraph>
        <Link href="/en/overview">Overview</Link>
      </>
    </PensionsDashboardLayout>
  );
};

export default Page;
