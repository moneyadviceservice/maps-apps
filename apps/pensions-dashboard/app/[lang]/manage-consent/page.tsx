import { Metadata, NextPage } from 'next';
import { PensionsDashboardLayout } from '../../../layouts/PensionsDashboardLayout';
import { SITE_TITLE } from '../../../utils/constants';
import { Paragraph } from '@maps-digital/ui/components/Paragraph';

const title = 'Manage Consent';

export const generateMetadata = async (): Promise<Metadata> => {
  return {
    title: `${title} | ${SITE_TITLE}`,
  };
};

const Page: NextPage = async () => {
  return (
    <PensionsDashboardLayout title={title}>
      <Paragraph className="text-center py-24">
        A page to manage consent
      </Paragraph>
    </PensionsDashboardLayout>
  );
};

export default Page;
