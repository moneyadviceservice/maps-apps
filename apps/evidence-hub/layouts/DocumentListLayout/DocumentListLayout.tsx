import { DocumentSummary } from 'components/DocumentSummary';
import { SideNavigation } from 'components/SideNavigation';
import { twMerge } from 'tailwind-merge';
import { DocumentTemplate, Tag } from 'types/@adobe/page';
import { v4 } from 'uuid';

import { H2 } from '@maps-react/common/components/Heading';

interface DocumentListLayoutProps {
  documents: DocumentTemplate[];
  lang: string;
  tags: {
    label: string;
    tags: Tag[];
  }[];
  query: Record<string, string | string[] | undefined>;
}

export const DocumentListLayout = ({
  documents,
  lang,
  tags,
  query,
}: DocumentListLayoutProps) => {
  return (
    <div
      className={twMerge(['mt-8 lg:mt-12 flex flex-col lg:flex-row lg:gap-8'])}
    >
      <SideNavigation tags={tags} query={query} className="flex-shrink-0" />
      <div className="flex-1">
        <div className="max-w-4xl">
          {documents.length === 0 ? (
            <div className="py-12">
              <H2 className="mb-16">
                Weâ€™re sorry, no results have been found for your search.
              </H2>
              <p className="mb-6">Improve your search by:</p>
              <ul className="list-disc list-inside">
                <li>double-checking your spelling.</li>
                <li>using fewer filters.</li>
                <li>trying more general keywords</li>
                <li>trying different keywords.</li>
              </ul>
            </div>
          ) : (
            <div className="space-y-8">
              <div className="mb-8">
                <p className="text-xl font-bold text-gray-600">
                  {documents.length} document
                  {documents.length === 1 ? '' : 's'} found
                </p>
              </div>

              {documents.map((doc: DocumentTemplate) => (
                <DocumentSummary key={v4()} doc={doc} lang={lang} />
              ))}
            </div>
          )}
        </div>
      </div>
    </div>
  );
};
