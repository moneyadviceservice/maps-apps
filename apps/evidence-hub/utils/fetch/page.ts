import { DocumentTemplate } from 'types/@adobe/page';

import { aemHeadlessClient } from '@maps-react/utils/aemHeadlessClient';

type PageError = {
  error: boolean;
};

export const fetchPage = async (
  lang: string,
  slug: string,
): Promise<DocumentTemplate | PageError> => {
  try {
    const queryName = `evidence-hub/get-page-${lang};slug=${slug}`;

    const {
      data: {
        pageTemplateList: { items: pageTemplate },
      },
    } = await aemHeadlessClient.runPersistedQuery(queryName);

    if (pageTemplate.length === 0) {
      return {
        error: true,
      } as PageError;
    }

    const result = pageTemplate[0];

    return result;
  } catch (error) {
    console.error('failed to fetch page ', error);
    return {
      error: true,
    } as PageError;
  }
};
