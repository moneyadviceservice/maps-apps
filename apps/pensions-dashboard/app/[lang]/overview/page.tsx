import { Metadata, NextPage } from 'next';
import { twMerge } from 'tailwind-merge';
import { PensionsDashboardLayout } from '../../../layouts/PensionsDashboardLayout';
import { SITE_TITLE, NUMBER_WORDS } from '../../../utils/constants';
import { getPensionsOverview } from '../../../utils/fetch/get-pensions-overview';
import { Button } from '@maps-digital/ui/components/Button';
import { Callout, CalloutVariant } from '@maps-digital/ui/components/Callout';
import { Heading } from '@maps-digital/ui/components/Heading';
import { Icon, IconType } from '@maps-digital/ui/components/Icon';

const title = 'Pensions found';

export const generateMetadata = async (): Promise<Metadata> => {
  return {
    title: `${title} | ${SITE_TITLE}`,
  };
};

const Page: NextPage = async () => {
  const {
    totalPensions,
    incompletePensions,
    confirmedPensions,
    unconfirmedPensions,
  } = await getPensionsOverview();

  const calloutClasses = 'mb-6 p-8';
  const calloutHeadingClasses = 'mb-4';
  const calloutItemClasses = 'flex gap-2 mt-4';
  const calloutCtaClasses = 'mb-12';

  const numberToWords = (num: number) =>
    num > NUMBER_WORDS.length ? num : NUMBER_WORDS[num].en;

  return (
    <PensionsDashboardLayout title={title}>
      <Heading level="h2" className="mb-12">
        We found {numberToWords(totalPensions)} pension
        {totalPensions === 1 ? '' : 's'}
      </Heading>

      <div className="grid grid-cols-3 gap-16">
        <div className="col-span-3 lg:col-span-2">
          {confirmedPensions.length > 0 && (
            <>
              <Callout
                variant={CalloutVariant.POSITIVE}
                className={twMerge(calloutClasses)}
              >
                <Heading level="h6" className={twMerge(calloutHeadingClasses)}>
                  Pensions with complete information
                </Heading>
                <ul>
                  {confirmedPensions.map(
                    ({
                      externalAssetId,
                      pensionType,
                      schemeName,
                      pensionAdministrator: { name },
                    }) => (
                      <li
                        key={externalAssetId}
                        className={twMerge(calloutItemClasses)}
                      >
                        <Icon type={IconType.TICK_GREEN} />
                        {pensionType === 'SP' ? schemeName : name}
                      </li>
                    ),
                  )}
                </ul>
              </Callout>
              <Button
                as="a"
                href="/en/your-pensions"
                className={twMerge(calloutCtaClasses)}
              >
                See pension breakdown
              </Button>
            </>
          )}

          {incompletePensions.length > 0 && (
            <>
              <Callout
                variant={CalloutVariant.WARNING}
                className={twMerge(calloutClasses)}
              >
                <Heading level="h6" className={twMerge(calloutHeadingClasses)}>
                  Pensions waiting for more information from your pension
                  providers
                </Heading>
                <ul>
                  {incompletePensions.map(
                    ({ externalAssetId, pensionAdministrator: { name } }) => (
                      <li
                        key={externalAssetId}
                        className={twMerge(calloutItemClasses)}
                      >
                        <Icon type={IconType.EDIT} />
                        {name}
                      </li>
                    ),
                  )}
                </ul>
              </Callout>
              <Button
                as="a"
                href="/en/pending-pensions"
                className={twMerge(calloutCtaClasses)}
              >
                View pending pensions
              </Button>
            </>
          )}

          {unconfirmedPensions.length > 0 && (
            <>
              <Callout
                variant={CalloutVariant.NEGATIVE}
                className={twMerge(calloutClasses)}
              >
                <Heading level="h6" className={twMerge(calloutHeadingClasses)}>
                  Pensions that are a possible match with your details
                </Heading>
                <ul>
                  {unconfirmedPensions.map(
                    ({ externalAssetId, pensionAdministrator: { name } }) => (
                      <li
                        key={externalAssetId}
                        className={twMerge(calloutItemClasses)}
                      >
                        <Icon
                          type={IconType.WARNING}
                          className="scale-50 m-[-12px]"
                        />
                        {name}
                      </li>
                    ),
                  )}
                </ul>
              </Callout>
              <Button
                as="a"
                href="/en/pensions-that-need-action"
                className={twMerge(calloutCtaClasses)}
              >
                Review pensions that need action
              </Button>
            </>
          )}
        </div>
      </div>
    </PensionsDashboardLayout>
  );
};

export default Page;
