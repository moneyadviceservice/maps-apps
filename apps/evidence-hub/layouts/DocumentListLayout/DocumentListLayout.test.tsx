import { DocumentTemplate, Tag } from 'types/@adobe/page';
import { render, screen } from '@testing-library/react';

import { DocumentListLayout } from './DocumentListLayout';

import '@testing-library/jest-dom';

jest.mock('components/DocumentSummary', () => ({
  DocumentSummary: ({ doc }: { doc: DocumentTemplate }) => (
    <div data-testid="document-summary">{doc.title}</div>
  ),
}));
jest.mock('components/SideNavigation', () => ({
  SideNavigation: ({ tags }: { tags: { label: string; tags: Tag[] }[] }) => (
    <nav data-testid="side-navigation">{tags.length} tags</nav>
  ),
}));

const baseDoc: DocumentTemplate = {
  seoTitle: 'SEO Title',
  seoDescription: 'SEO description',
  title: 'Sample Doc',
  slug: 'sample-doc',
  publishDate: '2023-01-01T00:00:00Z',
  sections: [],
  contactInformation: { json: [] },
  clientGroup: [],
  topic: [],
  countryOfDelivery: [],
  links: [],
};

describe('DocumentListLayout', () => {
  it('renders empty state when no documents', () => {
    render(
      <DocumentListLayout
        documents={[]}
        lang="en"
        tags={[{ label: 'Category', tags: [] }]}
        query={{}}
      />,
    );

    expect(
      screen.getByText(
        /Weâ€™re sorry, no results have been found for your search./i,
      ),
    ).toBeInTheDocument();
    expect(screen.getByTestId('side-navigation')).toBeInTheDocument();
  });

  it('renders list of documents when provided', () => {
    const { container, getByText, getByTestId } = render(
      <DocumentListLayout
        documents={[baseDoc]}
        lang="en"
        tags={[{ label: 'Category', tags: [] }]}
        query={{}}
      />,
    );

    expect(getByText(/1 document found/i)).toBeInTheDocument();
    expect(getByTestId('document-summary')).toHaveTextContent('Sample Doc');
    expect(container).toMatchSnapshot();
  });
});
