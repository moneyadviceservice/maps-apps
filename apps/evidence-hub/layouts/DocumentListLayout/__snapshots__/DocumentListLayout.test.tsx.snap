// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentListLayout renders list of documents when provided 1`] = `
<div>
  <div
    class="mt-8 lg:mt-12 flex flex-col lg:flex-row lg:gap-8"
  >
    <nav
      data-testid="side-navigation"
    >
      1
       tags
    </nav>
    <div
      class="flex-1"
    >
      <div
        class="max-w-4xl"
      >
        <div
          class="space-y-8"
        >
          <div
            class="mb-8"
          >
            <p
              class="text-xl font-bold text-gray-600"
            >
              1
               document
               found
            </p>
          </div>
          <div
            data-testid="document-summary"
          >
            Sample Doc
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
